### PR TITLE
Use a custom docker image for testing

### PR DESCRIPTION
**Pull request by m-kuhn**
This is exactly the same qgis version as osgeo4a and other dependencies are newer than qgis/qgis at the time of writing.

Will need to be restarted when the docker is ready.

### CODE REVIEW

File: sdk.conf [hunk 1]
```diff
@@ -1 +1 @@
-osgeo4a_version=20200705
+osgeo4a_version=linux_build
```
Comment (suricactus) [inline sdk.conf:1]: This will mean the latest version?

**Comment by suricactus** [comment]
Before merging, for some reason `vertexmodeltest` is failing?

File: sdk.conf (hunk 1 above)
Comment (m-kuhn) [reply sdk.conf:1]: We push:
 - from tags/releases on OSGeo4A an image with the tag
 - from branch pushes on OSGeo4A an image with the branch name (that's what happened here)

The next "release" will be available as soon as it's done

**Comment by m-kuhn** [comment]
> Before merging, for some reason `vertexmodeltest` is failing?

Probably related to the update from proj 4 to proj 6.

File: sdk.conf (hunk 1 above)
Comment (m-kuhn) [reply sdk.conf:1]:
```suggestion
osgeo4a_version=20200731
```

About to be released

**Comment by 3nids** [comment]
> Probably related to the update from proj 4 to proj 6

Well, apparently grids are missing in proj.
I guess we need to add https://github.com/OSGeo/PROJ-data but the release is about 500mb.

What do you think?

**Comment by m-kuhn** [comment]
I'm currently rewriting the tests to not depend on grids.
I would do the download of grids post installation when required.

**Comment by m-kuhn** [comment]
@suricactus I guess we can download the failing travis test after this one is integrated and now working on gh workflows?

**Comment by suricactus** [comment]
Yes, the tests are already running properly via github workflows.

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1179-f9525080ad8dc5aed953e84c3f1bd8001d5a881b-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1179-f9525080ad8dc5aed953e84c3f1bd8001d5a881b-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1179-f9525080ad8dc5aed953e84c3f1bd8001d5a881b-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1179-f9525080ad8dc5aed953e84c3f1bd8001d5a881b-x86_64.apk)

**Pull request merged**